### PR TITLE
Upgrade to SDL2.0.20

### LINKED ISSUE #267: SDL_GameControllerHasRumble is undefine
In src/game_controller.cpp line 324
SDL_GameControllerHasRumble is undefine
Fail to build joy

### PR DESCRIPTION
**Pull request by roncapat**
Fix #267

### CODE REVIEW

File: sdl2_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -14,10 +14,10 @@ endif()
 
 if(NOT SDL2_FOUND)
   include(ExternalProject)
-  ExternalProject_Add(SDL2-2.0.14
-    PREFIX SDL2-2.0.12
-    URL https://www.libsdl.org/release/SDL2-2.0.14.tar.gz
-    URL_MD5 76ed4e6da9c07bd168b2acd9bfefab1b
+  ExternalProject_Add(SDL2-2.0.18
```
Comment (clalancette) [inline sdl2_vendor/CMakeLists.txt:17]: If we are going to update it, I'd rather we update to 2.0.20, as that is the version that is shipped in Ubuntu 22.04.  Do you mind doing that?

**Review by clalancette — CHANGES_REQUESTED** [review]

File: sdl2_vendor/CMakeLists.txt (hunk 1 above)
Comment (roncapat) [reply sdl2_vendor/CMakeLists.txt:17]: Ah yes, of course :)

**Comment by roncapat** [comment]
@clalancette I noticed a problem with the `sdl2_vendor/sdl2-windows-add-vcruntime.patch` patch not being applied. I'll do some checks and report.

**Comment by roncapat** [comment]
Should be fixed now. I warn you that it seems to me that during CI the SDL doesn't get built from sources (it may already be present in Ubuntu packages). Maybe `FORCE_BUILD_VENDOR_PKG` is not set?

**Comment by clalancette** [comment]
> Maybe `FORCE_BUILD_VENDOR_PKG` is not set?

Yeah, it is not: https://github.com/ros-drivers/joystick_drivers/blob/ros2/.github/workflows/rolling-build-ci.yaml

The vendoring of this package was mostly meant for Windows; on Linux, we generally have a packaged SDL available (though I guess in this case, it might be too old).  One thing we could consider would be to build it if the SDL version is not new enough to have the game controller interface.

**Review by clalancette — APPROVED** [review]
Looks good to me, thanks for the update.

**Pull request merged**